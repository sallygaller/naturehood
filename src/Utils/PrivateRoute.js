import React from "react";
import { Route, Redirect } from "react-router-dom";
import TokenService from "../services/token-service";

export default function PrivateRoute({ component, ...props }) {
  const Component = component;
  return (
    <Route
      {...props}
      render={(componentProps) =>
        TokenService.hasAuthToken() ? (
          (console.log({ componentProps }, { props }),
          (<Component {...componentProps} {...props} />))
        ) : (
          <Redirect
            to={{
              pathname: "/login",
              state: { from: componentProps.location },
            }}
          />
        )
      }
    />
  );
}
