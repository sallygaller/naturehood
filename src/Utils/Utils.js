import moment from "moment";

export function dateFormat(observation) {
  const dateFormat = moment(observation.date).format("L");
  return dateFormat;
}

export function timeFormat(observation) {
  const timeFormat = moment(observation.time, "hh:mm:ss").format("hh:mm a");
  return timeFormat;
}

export const speciesTypes = [
  { id: 1, type: "Mammal" },
  { id: 2, type: "Fish" },
  { id: 3, type: "Bird" },
  { id: 4, type: "Arthropod" },
  { id: 5, type: "Amphibian" },
  { id: 6, type: "Reptile" },
];

export const testObservations = [
  {
    id: 1,
    species: "Robin",
    type: "Bird",
    date: "2020-12-25T08:00:00.000Z",
    time: "08:30:00",
    description: "I saw a robin at my feeder",
    lat: 45.593,
    lng: -122.755,
    date_added: "2020-12-25T09:28:32.615Z",
    neighbor: 1,
  },
  {
    id: 2,
    species: "Fox",
    type: "Mammal",
    date: "2020-12-15T08:00:00.000Z",
    time: "04:30:00",
    description: "I saw a mother fox and her cubs. The mother hissed at me.",
    lat: 47.593,
    lng: -122.755,
    date_added: "2020-12-15T16:28:32.615Z",
    neighbor: 1,
  },
  {
    id: 3,
    species: "Raccoon",
    type: "Mammal",
    date: "2020-12-05T08:00:00.000Z",
    time: "23:30:00",
    description: "A group of raccoons chased me down the street.",
    lat: 49.593,
    lng: -122.755,
    date_added: "2020-12-06T09:28:32.615Z",
    neighbor: 1,
  },
  {
    id: 4,
    species: "Skunk",
    type: "Mammal",
    date: "2020-12-08T08:00:00.000Z",
    time: "07:30:00",
    description: "I saw a skunk under my deck.",
    lat: 51.593,
    lng: -123.755,
    date_added: "2020-12-08T10:15:32.615Z",
    neighbor: 1,
  },
];
