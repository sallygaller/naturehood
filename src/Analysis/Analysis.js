import React from "react";
import { speciesTypes } from "../Utils/Utils";
import moment from "moment";

function countSpecies(speciesType, observations) {
  let type = speciesType.type;
  let results = 0;
  for (let i = 0; i < observations.length; i++) {
    if (observations[i].type === type) {
      results = results + 1;
    }
  }
  return results;
}

function dateLatest(observations) {
  let dates = observations.map((observation) => {
    return moment(observation.date);
  });
  let maxDate = moment.max(dates);
  let maxDateCreate = maxDate.creationData().input;
  let dateFormat = moment(maxDateCreate).format("L");
  return dateFormat;
}

export default function Analysis(props) {
  const observations = props.observations;
  return (
    <div className="Analysis">
      <p>
        Total observations: {observations.length} <br></br>
        Most recent observation: {dateLatest(observations)}
      </p>
      <ul>
        {speciesTypes.map((speciesType) => (
          <li key={speciesType.id}>
            {countSpecies(speciesType, observations)} {speciesType.type}
            {speciesType.type === "Fish" ? "" : "s"} seen
          </li>
        ))}
      </ul>
    </div>
  );
}
