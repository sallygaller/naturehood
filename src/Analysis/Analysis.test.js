import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Analysis from "./Analysis";
import { observations } from "../Utils/TestFiles";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <Analysis observations={observations} />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(<Analysis name="Analysis" observations={observations} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders the UI as expected with no observations", () => {
  const tree = renderer
    .create(<Analysis name="Analysis" observations={[]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
