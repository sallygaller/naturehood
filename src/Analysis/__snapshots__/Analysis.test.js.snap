// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Analysis"
>
  <p>
    Total observations: 
    3
     
    <br />
    Most recent observation: 
    12/25/2020
  </p>
  <ul>
    <li>
      <div
        className="SpeciesAnalysis"
      >
        2
         
        <img
          alt="Mammal"
          className="SpeciesAnalysis-img"
          src="mammal.jpeg"
        />
        seen
      </div>
    </li>
    <li>
      <div
        className="SpeciesAnalysis"
      >
        0
         
        <img
          alt="Fish"
          className="SpeciesAnalysis-img"
          src="fish.jpeg"
        />
        seen
      </div>
    </li>
    <li>
      <div
        className="SpeciesAnalysis"
      >
        1
         
        <img
          alt="Bird"
          className="SpeciesAnalysis-img"
          src="bird.jpeg"
        />
        seen
      </div>
    </li>
    <li>
      <div
        className="SpeciesAnalysis"
      >
        0
         
        <img
          alt="Arthropod"
          className="SpeciesAnalysis-img"
          src="insect.jpeg"
        />
        seen
      </div>
    </li>
    <li>
      <div
        className="SpeciesAnalysis"
      >
        0
         
        <img
          alt="Amphibian"
          className="SpeciesAnalysis-img"
          src="amphibian.jpeg"
        />
        seen
      </div>
    </li>
    <li>
      <div
        className="SpeciesAnalysis"
      >
        0
         
        <img
          alt="Reptile"
          className="SpeciesAnalysis-img"
          src="reptile.jpeg"
        />
        seen
      </div>
    </li>
  </ul>
</div>
`;

exports[`renders the UI as expected with no observations 1`] = `
<div
  className="Analysis"
>
  <p>
    Total observations: 
    0
     
    <br />
    Most recent observation: 
    01/29/2021
  </p>
  <ul>
    <li>
      <div
        className="SpeciesAnalysis"
      >
        0
         
        <img
          alt="Mammal"
          className="SpeciesAnalysis-img"
          src="mammal.jpeg"
        />
        seen
      </div>
    </li>
    <li>
      <div
        className="SpeciesAnalysis"
      >
        0
         
        <img
          alt="Fish"
          className="SpeciesAnalysis-img"
          src="fish.jpeg"
        />
        seen
      </div>
    </li>
    <li>
      <div
        className="SpeciesAnalysis"
      >
        0
         
        <img
          alt="Bird"
          className="SpeciesAnalysis-img"
          src="bird.jpeg"
        />
        seen
      </div>
    </li>
    <li>
      <div
        className="SpeciesAnalysis"
      >
        0
         
        <img
          alt="Arthropod"
          className="SpeciesAnalysis-img"
          src="insect.jpeg"
        />
        seen
      </div>
    </li>
    <li>
      <div
        className="SpeciesAnalysis"
      >
        0
         
        <img
          alt="Amphibian"
          className="SpeciesAnalysis-img"
          src="amphibian.jpeg"
        />
        seen
      </div>
    </li>
    <li>
      <div
        className="SpeciesAnalysis"
      >
        0
         
        <img
          alt="Reptile"
          className="SpeciesAnalysis-img"
          src="reptile.jpeg"
        />
        seen
      </div>
    </li>
  </ul>
</div>
`;
